Modules/BenchPress.Azure/Public/CustomAssertions/BeInLocation.ps1

```diff
@@ -31,12 +31,21 @@ function ShouldBeInLocation ($ActualValue, [string]$ExpectedValue, [switch] $Neg
     if ($Negate) { $succeeded = -not $succeeded }
     $failureMessage = "ConfirmResult is null or empty."
   }
+  elseif (-not $ActualValue.ResourceDetails.Location) {
```

Comment:
This method for checking whether a property exists will not work in strict mode, which the user can change (https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/set-strictmode?view=powershell-7.3#parameters). Alternatives could be:

`[bool]$ActualValue.ResourceDetails.PSObject.Properties[$Property]`
`[bool]$ActualValue.ResourceDetails.PSObject.Properties.Item($property)`

There are other ways, but according to https://stackoverflow.com/a/70492574/275796 those are the fastest.

Info on `.PSObject` can be found here:
https://learn.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-pscustomobject?view=powershell-7.3#working-with-properties
https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_intrinsic_members?view=powershell-7.3#object-views

Reply:
I don't know why I didn't just copy the if-statement above and use this to check for null or empty:
`$null -eq $ActualValue.ResourceDetails.Location -or '' -eq $ActualValue.ResourceDetails.Location`

That should work seeing as we use it as our standard checking.

I also tried the method above:
`$propertyName = 'Location'`
`[bool]$ActualValue.PSObject.Properties[$propertyName]`

 And they both pass the unit tests. Could go with either. First one is more understandable at first pass.

Reply:
I would say it depends on what is being tested. If we're testing whether a property exists, then the latter does that, the former does not. If we're testing whether a property has a value, then the inverse is true.

My understanding is that we're testing whether a property exists though. The property, if it exists, should always have a value from the Az PowerShell response, it would be a bug if it did not. And even if it did not, it would not match to the expected value and would fail the test as it should. With that said, my recommendation is to go with the latter, because it expresses the intent appropriately, and does not have any unwanted side effects.

Note, that your code in your comment is likely incorrect, unless I'm missing something. It should be `[bool]$ActualValue.ResourceDetails.PSObject.Properties[$propertyName]`

Reply:
I forgot one other thing. If you execute:

`$null -eq $ActualValue.ResourceDetails.Location -or '' -eq $ActualValue.ResourceDetails.Location`

in strict mode and the `Location` property does not exist, it will throw an exception. That was the point of the original comment.

Reply:
I'm okay with the second one. Added.